### PR TITLE
Filter out invalid prefixes and items from listing.

### PR DESCRIPTION
**Pull request by yuchenshi**
### Description

This filters out invalid items and prefixes from Firebase Storage list API results. Similar to production logic at go/firebase-storage-backend-valid-path (Googler only).

### Scenarios Tested

See tests added. Manually tested in Emulator UI too: created a folder, and it shows up in parent folder correctly. Inside the new folder, the view is empty with no placeholder showing up as files.

### CODE REVIEW

File: src/emulator/storage/apis/firebase.ts [hunk 1]
```diff
@@ -517,3 +518,28 @@ function setObjectHeaders(
     res.setHeader("Content-Language", metadata.contentLanguage);
   }
 }
+
+function validPrefix(prefix: string) {
```
Comment (tonyjhuang) [inline src/emulator/storage/apis/firebase.ts:522]: nit: isValidPrefix

Comment (tonyjhuang) [inline src/emulator/storage/apis/firebase.ts:522]: nit: return type?

File: scripts/storage-emulator-integration/tests.ts [hunk 2]
```diff
@@ -1710,6 +1721,63 @@ describe("Storage emulator", () => {
             items: [],
           });
         });
+
+        context("with folder placeholders", () => {
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/tests.ts:1725]: can you include a test that validates that invalid paths are filtered out of the list response please, e.g. `//foo/` `bar//` `/baz//qux/`

File: src/emulator/storage/apis/firebase.ts (hunk 1 above)
Comment (yuchenshi) [reply src/emulator/storage/apis/firebase.ts:522]: This is intentionally matching naming in storage backend (for easier cross-reference). I'm open to review a CL that renames it internally for consistency though.

Comment (tonyjhuang) [reply src/emulator/storage/apis/firebase.ts:522]: I don't think the two need to move in lockstep, we can clean up the internal version too but let's try to follow the style guide here

File: scripts/storage-emulator-integration/tests.ts (hunk 2 above)
Comment (yuchenshi) [reply scripts/storage-emulator-integration/tests.ts:1724]: Done.

File: src/emulator/storage/apis/firebase.ts (hunk 1 above)
Comment (yuchenshi) [reply src/emulator/storage/apis/firebase.ts]: Done.

**Review by tonyjhuang — APPROVED** [review]

File: src/emulator/storage/apis/firebase.ts (hunk 1 above)
Comment (yuchenshi) [reply src/emulator/storage/apis/firebase.ts]: Sure, renamed

**Pull request merged**